Create orgs

vincepmartin opened the pull request:
Work in progress.

Created code to manage Organizations.

Still needs testing/review.

vincepmartin committed:
updated the index.js for the organizations.

vincepmartin committed:
organizationsModel.js was refering to fieldModel.js. I do not see why we need this...

vincepmartin committed:
fixed spelling error on organizationsController.

vincepmartin committed:
tested and fixed all organizations api commands. all commands past IRL db test.

vincepmartin committed:
Removed some todos.

Andykmcc committed:
add .swp to gitignore

Andykmcc commented on test/api/v1/organizations/old/effortTypesController.spec.js:
```diff
@@ -0,0 +1,106 @@
+      const secondArgPassedToQuery = queryResolve.query.getCall(0).args[1];
+
+      assert.deepEqual(secondArgPassedToQuery, {id: guid});
+      queryStub.restore();
```
you shouldn't need this since there is an `afterEach` above this. if for some reason you do, it should be in an afterEach placed someplace else.

Andykmcc commented on test/api/v1/organizations/organizationsController.spec.js:
```diff
@@ -0,0 +1,103 @@
+  });
+
+  // TODO: Not sure this should be organization... Maybe it should be organizations?
+  describe('#getOrganizations', () => {
```
i don't think you need this, or at least not this check. i would just test that this function returns a promise. which in this case is a bit stupid since we have mocked the response to the only function called here so you are really just testing the mock but that won't be the case for long with there generic get functions so having a basic test stubbed out won't hurt.

so yeah, i'd change this to just test that`getOrganizations` returns a promise.

vincepmartin replied:
OK. I am going to re-check through the testing aspect of things. I'll probably need some help later on and have some questions. I just saw it passed the git commit tests and thought it was ok.

Andykmcc replied:
weird, it shouldn't have let you commit with failing tests, unless you run `git commit -n` which ignores the commit hooks. you might need to delete you `node_moduels` directory and `npm install` again. 

i'm thinking of changing the tests hook so it blocks you from pushing, but not committing. my thinking on that is that committing often is good (pushing often is good too) so i want to make committing as painless as possible.

Andykmcc replied:
when you are running tests via `npm test` or `npm run test:watch` be sure to stop Neo4j. otherwise you run the chance of your tests hitting your DB and that is bad. also since there is no DB in the cloud you might end up with tests passing on your local that fail in Travis.CI

Andykmcc reviewed (commented):
overall this looks great. tests are still failing but i know you are still working on those. 

i added .swp files to the .gitignore at the project root. you need to remove the swp file in the org tests directory from git on your end so it isn't tracked going forward. see this link for details, https://stackoverflow.com/questions/1274057/how-to-make-git-forget-about-a-file-that-was-tracked-but-is-now-in-gitignore

also, does all the `proxyquire` stuff make sense? it is kind of hacky. and i haven't documented it anywhere. i'm happy to explain why I'm using.

vincepmartin committed:
fixing index.js

vincepmartin committed:
fixing deleted files.

vincepmartin committed:
fixing .gitignore

vincepmartin committed:
actually fixing .gitignore this time...

Andykmcc commented:
FYI, it looks like your tests are passing in Travis. you just need to update your branch with that has changed in master.

Andykmcc commented:
sorry, clicked the wrong button, i didn't mean to close this.

vincepmartin committed:
Merge branch 'master' into create-orgs

vincepmartin committed:
In progress: Adding code to delete orgs along w/ all attached nodes except users.

vincepmartin committed:
update

vincepmartin committed:
Merge branch 'master' into create-orgs
Andy made some changes. I'm going to pull them in to see if it fixes my
problem of getting null responses from the API on get requests.

Andykmcc commented on src/api/v1/organizations/organizationsController.js:
```diff
@@ -0,0 +1,92 @@
+
+  // Removes all Users employed by the organization.
+  let removeEmployeesResults = neo4jHelpers.query(`
+    MATCH (o:Organization)-[r:Employs]->(u:User) DELETE r
```
can you put each keyword (`MATCH`, `DELETE`, etc...) on a new line

Andykmcc commented on src/api/v1/organizations/organizationsController.js:
```diff
@@ -0,0 +1,92 @@
+  // Removes all Users employed by the organization.
+  let removeEmployeesResults = neo4jHelpers.query(`
+    MATCH (o:Organization)-[r:Employs]->(u:User) DELETE r
+      WHERE o.id = {id}
```
you don't need to do this as a WHERE clause. you can match the org like this...
```
MATCH (o:Organization { id: {id} })
```

Andykmcc commented on src/api/v1/organizations/organizationsController.js:
```diff
@@ -0,0 +1,92 @@
+  }
+
+  // Removes all Users employed by the organization.
+  let removeEmployeesResults = neo4jHelpers.query(`
```
this can be `const` since there is no need to reassign it.

Andykmcc commented:
i merged my change to the query handleSuccess parser so that should fix a problem you were having with this yesterday at the coffee shop.

vincepmartin committed:
Fixing constant name.

vincepmartin committed:
Merge branch 'master' into create-orgs

vincepmartin committed:
Altered the cypher queries so that they do not use match. This works
fine in most cases, however, in the delete methods it does not.

I have to rethink how I am doing these. Going to look into some more
cypher stuff tomorrow.

vincepmartin committed:
in progress, organizationsController debugging.

Andykmcc commented:
closing for now to clean up by github assigned ticekts
